db/db.go

```diff
@@ -0,0 +1,129 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+
+	sq "github.com/Masterminds/squirrel"
+)
+
+// DB has all the tooling for talking to postgres
+type DB struct {
+	psql sq.StatementBuilderType
+}
+
+// New creates a DB instance
+func New(psql sq.StatementBuilderType) *DB {
+	return &DB{
+		psql: psql,
+	}
+}
+
+// Post represents a post in JSON
+type Post struct {
+	ID    int    `json:"id"`
+	Title string `json:"title"`
+	Body  string `json:"body"`
+}
+
+// GetPosts fetches all posts
+func (db *DB) GetPosts(ctx context.Context) (_ []Post, retErr error) {
+	posts := []Post{}
+
+	rows, err := db.psql.Select("*").From("posts").Query()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		retErr = rows.Close()
+	}()
+
+	for rows.Next() {
+		var p Post
+
+		if err := rows.Scan(&p.ID, &p.Title, &p.Body); err != nil {
+			return nil, err
+		}
+
+		posts = append(posts, p)
+	}
+
+	if rows.Err() != nil {
+		return nil, err
+	}
+
+	return posts, nil
+}
+
+// GetPost fetches a post by ID
+func (db *DB) GetPost(ctx context.Context, id int) (Post, error) {
+	var p Post
+
+	if err := db.psql.
+		Select("id", "title", "body").
+		From("posts").Where(sq.Eq{"id": id}).
+		QueryRow().
+		Scan(&p.ID, &p.Title, &p.Body); err != nil {
+		return Post{}, err
+	}
+
+	return p, nil
+}
+
+// CreatePost creates a new post
+func (db *DB) CreatePost(ctx context.Context, p *Post) error {
+	err := db.psql.
+		Insert("posts").
+		Columns("title", "body").
+		Values(p.Title, p.Body).
```

Comment:
This is fine for small columns, but I actually prefer to use `SetMap` generally, as it makes the mapping between column and value explicit:
```suggestion
		SetMap(map[string]interface{}[
			"title": p.Title,
			"body", p.Body,
		}).
```
They're logically equivalent, it's up to you which one you prefer, this is only a suggestion.